### PR TITLE
create book

### PR DESCRIPTION
**Pull request by sujoydatta**

### CODE REVIEW

File: app/views/shared/_error_messages.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(object.errors.count, "error") %>.
```
Comment (MuhammadTamzid) [inline app/views/shared/_error_messages.html.erb]: i18n

**Comment by MuhammadTamzid** [comment]
Ok

File: config/routes.rb [hunk 2]
```diff
@@ -4,5 +4,9 @@
     get "home" => "static_pages#home"
     get "about" => "static_pages#about"
     get "contact" => "static_pages#contact"
+
+    namespace :admin do
```
Comment (dieunb) [inline config/routes.rb]: admin root path

File: app/views/admin/books/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<h1><%= t "add_book_text" %></h1>
+<div class="row">
+  <div class="col-md-8 col-md-offset-2">
+    <%= nested_form_for [:admin, @book] do |f| %>
+      <%= render "shared/error_messages", object: f.object %>
+
+      <%= f.label :category %>
+      <%= f.collection_select :category_id, Category.all, :id, :name,
```
Comment (dieunb) [inline app/views/admin/books/new.html.erb]: move `Category.all` to controller

File: app/controllers/admin/books_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class Admin::BooksController < ApplicationController
+
+  def new
+    @book = Book.new
```
Comment (dieunb) [inline app/controllers/admin/books_controller.rb]: after implement `cancancan`, pls fix it.

File: app/controllers/admin/books_controller.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def create
+    @book = Book.new book_params
```
Comment (dieunb) [inline app/controllers/admin/books_controller.rb]: after implement `cancancan`, pls fix it.

File: app/views/shared/_error_messages.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t "form_contains" %> <%= pluralize(object.errors.count, "error") %>.
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: remove `()`
```
<%= pluralize object.errors.count, "error" %>.
```

File: app/views/shared/_error_messages.html.erb [hunk 7]
```diff
@@ -1,12 +1,12 @@
 <% if object.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
-      <%= t "error_frm" %> <%= pluralize(object.errors.count, "error") %>.
+      <%= t "form_contains" %> <%= pluralize(object.errors.count, "error") %>.
```
Comment (MuhammadTamzid) [inline app/views/shared/_error_messages.html.erb]: remove ()

**Pull request merged**